### PR TITLE
Fix: Watcher API watcher_get_settings clears product header

### PR DESCRIPTION
**Pull request by ldematte**
Current code do stashing and un-stashing of the context to ignore warning headers about accessing a system index. However, this ignores _all_ header, omitting the `X-elastic-product header` from the response.

This PR removes the stashing/un-stashing and instead calls the `concreteIndexNames` variant that allows system index access (which does not generate the warning), `concreteIndexNamesWithSystemIndexAccess`

Closes https://github.com/elastic/elasticsearch/issues/102928

### CODE REVIEW

File: test/yaml-rest-runner/src/main/java/org/elasticsearch/test/rest/yaml/section/DoSection.java [hunk 1]
```diff
@@ -363,7 +363,7 @@ public void execute(ClientYamlTestExecutionContext executionContext) throws IOEx
             final String testPath = executionContext.getClientYamlTestCandidate() != null
                 ? executionContext.getClientYamlTestCandidate().getTestPath()
                 : null;
-            if (executionContext.esVersion().major == Version.V_7_17_0.major && executionContext.esVersion().after(Version.V_7_17_1)) {
+            if (executionContext.esVersion().after(Version.V_7_17_1)) {
```
Comment (ldematte) [inline test/yaml-rest-runner/src/main/java/org/elasticsearch/test/rest/yaml/section/DoSection.java:366]: This is fixed in a separate PR, will remove it in the final revision; it's here to check the fix works :)

**Comment by ldematte** [comment]
`GET _watcher/settings` is now fixed, but the same approach needs to be extended to other _watcher APIs too. Will push another commit.

**Review by gmarouli — APPROVED** [review]
LGTM, thank you @ldematte !

**Comment by ldematte** [comment]
Thank you Mary!

**Pull request merged**